angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete.component.ts

```diff
@@ -274,6 +274,17 @@ export class SprkAutocompleteComponent
     this.isOpen = true;
   }
 
+  /**
+   * If itemSelectedEvent is specified, also set that as the click event on each result
+   */
+  bindClickEvents(): void {
+    if (this.itemSelectedEvent) {
+      this.resultItems.forEach((element) => {
+        element.clickedEvent = this.itemSelectedEvent;
+      });
+    }
+  }
```

Comment:
I extracted the click event logic into this function. The function is called once when the component initializes and is called again any time the `resultItems` changs.

Reply:
I was wondering where `clickedEvent` came from because it doesn't show up anywhere else on the file.  On chat I learned it's an input of `SprkAutocompleteResultDirective`, and here, it's in the form of `resultItems`. If I'm understanding correctly this successfully connects to an event emitter of `SprkAutocompleteResultDirective`:

![image](https://user-images.githubusercontent.com/4342363/121210197-78d00a80-c849-11eb-90d2-06ebd70c3631.png)